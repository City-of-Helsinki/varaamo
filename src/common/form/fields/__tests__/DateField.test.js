import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { globalDateMock } from '../../../../../app/utils/testUtils';
import { UntranslatedDateField as DateField } from '../DateField';

describe('DateField', () => {
  globalDateMock();
  test('renders correctly', () => {
    const props = {
      onChange: jest.fn(),
      label: 'foo',
      id: 'foo',
      value: new Date(2019, 8, 9),
      locale: 'en',
    };

    const wrapper = shallow(
      <DateField {...props} />,
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
