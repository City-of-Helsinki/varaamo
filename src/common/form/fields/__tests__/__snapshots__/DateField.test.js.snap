// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateField renders correctly 1`] = `
<div
  className="app-DateField"
>
  <FormGroup
    bsClass="form-group"
    onClick={[Function]}
  >
    <ControlLabel
      bsClass="control-label"
      className="app-DateField__label"
      srOnly={false}
    >
      foo
    </ControlLabel>
    <InputGroup
      bsClass="input-group"
    >
      <InputGroupAddon
        bsClass="input-group-addon"
        className="app-DateField__input"
      >
        <img
          alt=""
          className="app-DateField__icon"
          src="test-file-stub"
        />
        <span
          className="value"
        >
          10.12.2017
        </span>
      </InputGroupAddon>
      <InputGroupAddon
        bsClass="input-group-addon"
        className="app-DateField__triangle"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="triangle-bottom"
        />
      </InputGroupAddon>
    </InputGroup>
  </FormGroup>
  <Overlay
    animation={false}
    container={
      UntranslatedDateField {
        "context": {},
        "onChange": [Function],
        "props": {
          "id": "foo",
          "label": "foo",
          "locale": "en",
          "onChange": [MockFunction],
          "value": 2017-12-09T22:00:00.000Z,
        },
        "refs": {},
        "state": {
          "isOpen": false,
        },
        "updater": Updater {
          "_callbacks": [],
          "_renderer": ReactShallowRenderer {
            "_context": {},
            "_currentlyRenderingComponent": null,
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useState": [Function],
            },
            "_element": <UntranslatedDateField
              id="foo"
              label="foo"
              locale="en"
              onChange={[MockFunction]}
              value={2017-12-09T22:00:00.000Z}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_previousComponentIdentity": null,
            "_renderPhaseUpdates": null,
            "_rendered": <div
              className="app-DateField"
            >
              <FormGroup
                bsClass="form-group"
                onClick={[Function]}
              >
                <ControlLabel
                  bsClass="control-label"
                  className="app-DateField__label"
                  srOnly={false}
                >
                  foo
                </ControlLabel>
                <InputGroup
                  bsClass="input-group"
                >
                  <InputGroupAddon
                    bsClass="input-group-addon"
                    className="app-DateField__input"
                  >
                    <img
                      alt=""
                      className="app-DateField__icon"
                      src="test-file-stub"
                    />
                    <span
                      className="value"
                    >
                      10.12.2017
                    </span>
                  </InputGroupAddon>
                  <InputGroupAddon
                    bsClass="input-group-addon"
                    className="app-DateField__triangle"
                  >
                    <Glyphicon
                      bsClass="glyphicon"
                      glyph="triangle-bottom"
                    />
                  </InputGroupAddon>
                </InputGroup>
              </FormGroup>
              <Overlay
                animation={false}
                container={[Circular]}
                onHide={[Function]}
                placement="bottom"
                rootClose={true}
                show={false}
              >
                <DatePickerWrapper>
                  <DayPicker
                    canChangeMonth={true}
                    captionElement={
                      <Caption
                        classNames={
                          {
                            "body": "DayPicker-Body",
                            "caption": "DayPicker-Caption",
                            "container": "DayPicker",
                            "day": "DayPicker-Day",
                            "disabled": "disabled",
                            "footer": "DayPicker-Footer",
                            "interactionDisabled": "DayPicker--interactionDisabled",
                            "month": "DayPicker-Month",
                            "months": "DayPicker-Months",
                            "navBar": "DayPicker-NavBar",
                            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                            "outside": "outside",
                            "selected": "selected",
                            "today": "today",
                            "todayButton": "DayPicker-TodayButton",
                            "week": "DayPicker-Week",
                            "weekNumber": "DayPicker-WeekNumber",
                            "weekday": "DayPicker-Weekday",
                            "weekdays": "DayPicker-Weekdays",
                            "weekdaysRow": "DayPicker-WeekdaysRow",
                            "wrapper": "DayPicker-wrapper",
                          }
                        }
                        localeUtils={
                          {
                            "formatDay": [Function],
                            "formatMonthTitle": [Function],
                            "formatWeekdayLong": [Function],
                            "formatWeekdayShort": [Function],
                            "getFirstDayOfWeek": [Function],
                            "getMonths": [Function],
                          }
                        }
                      />
                    }
                    classNames={
                      {
                        "body": "DayPicker-Body",
                        "caption": "DayPicker-Caption",
                        "container": "DayPicker",
                        "day": "DayPicker-Day",
                        "disabled": "disabled",
                        "footer": "DayPicker-Footer",
                        "interactionDisabled": "DayPicker--interactionDisabled",
                        "month": "DayPicker-Month",
                        "months": "DayPicker-Months",
                        "navBar": "DayPicker-NavBar",
                        "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                        "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                        "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                        "outside": "outside",
                        "selected": "selected",
                        "today": "today",
                        "todayButton": "DayPicker-TodayButton",
                        "week": "DayPicker-Week",
                        "weekNumber": "DayPicker-WeekNumber",
                        "weekday": "DayPicker-Weekday",
                        "weekdays": "DayPicker-Weekdays",
                        "weekdaysRow": "DayPicker-WeekdaysRow",
                        "wrapper": "DayPicker-wrapper",
                      }
                    }
                    enableOutsideDaysClick={true}
                    fixedWeeks={false}
                    initialMonth={2017-12-09T22:00:00.000Z}
                    labels={
                      {
                        "nextMonth": "Next Month",
                        "previousMonth": "Previous Month",
                      }
                    }
                    locale="en"
                    localeUtils={
                      {
                        "formatDate": [Function],
                        "formatDay": [Function],
                        "formatMonthTitle": [Function],
                        "formatWeekdayLong": [Function],
                        "formatWeekdayShort": [Function],
                        "getFirstDayOfWeek": [Function],
                        "getMonths": [Function],
                        "parseDate": [Function],
                      }
                    }
                    navbarElement={
                      <Navbar
                        classNames={
                          {
                            "body": "DayPicker-Body",
                            "caption": "DayPicker-Caption",
                            "container": "DayPicker",
                            "day": "DayPicker-Day",
                            "disabled": "disabled",
                            "footer": "DayPicker-Footer",
                            "interactionDisabled": "DayPicker--interactionDisabled",
                            "month": "DayPicker-Month",
                            "months": "DayPicker-Months",
                            "navBar": "DayPicker-NavBar",
                            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                            "outside": "outside",
                            "selected": "selected",
                            "today": "today",
                            "todayButton": "DayPicker-TodayButton",
                            "week": "DayPicker-Week",
                            "weekNumber": "DayPicker-WeekNumber",
                            "weekday": "DayPicker-Weekday",
                            "weekdays": "DayPicker-Weekdays",
                            "weekdaysRow": "DayPicker-WeekdaysRow",
                            "wrapper": "DayPicker-wrapper",
                          }
                        }
                        dir="ltr"
                        labels={
                          {
                            "nextMonth": "Next Month",
                            "previousMonth": "Previous Month",
                          }
                        }
                        showNextButton={true}
                        showPreviousButton={true}
                      />
                    }
                    numberOfMonths={1}
                    onDayClick={[Function]}
                    pagedNavigation={false}
                    renderDay={[Function]}
                    renderWeek={[Function]}
                    reverseMonths={false}
                    selectedDays={2017-12-09T22:00:00.000Z}
                    showOutsideDays={true}
                    showWeekDays={true}
                    showWeekNumbers={true}
                    tabIndex={0}
                    weekdayElement={<Weekday />}
                  />
                </DatePickerWrapper>
              </Overlay>
            </div>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    onHide={[Function]}
    placement="bottom"
    rootClose={true}
    show={false}
  >
    <DatePickerWrapper>
      <DayPicker
        canChangeMonth={true}
        captionElement={
          <Caption
            classNames={
              {
                "body": "DayPicker-Body",
                "caption": "DayPicker-Caption",
                "container": "DayPicker",
                "day": "DayPicker-Day",
                "disabled": "disabled",
                "footer": "DayPicker-Footer",
                "interactionDisabled": "DayPicker--interactionDisabled",
                "month": "DayPicker-Month",
                "months": "DayPicker-Months",
                "navBar": "DayPicker-NavBar",
                "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                "outside": "outside",
                "selected": "selected",
                "today": "today",
                "todayButton": "DayPicker-TodayButton",
                "week": "DayPicker-Week",
                "weekNumber": "DayPicker-WeekNumber",
                "weekday": "DayPicker-Weekday",
                "weekdays": "DayPicker-Weekdays",
                "weekdaysRow": "DayPicker-WeekdaysRow",
                "wrapper": "DayPicker-wrapper",
              }
            }
            localeUtils={
              {
                "formatDay": [Function],
                "formatMonthTitle": [Function],
                "formatWeekdayLong": [Function],
                "formatWeekdayShort": [Function],
                "getFirstDayOfWeek": [Function],
                "getMonths": [Function],
              }
            }
          />
        }
        classNames={
          {
            "body": "DayPicker-Body",
            "caption": "DayPicker-Caption",
            "container": "DayPicker",
            "day": "DayPicker-Day",
            "disabled": "disabled",
            "footer": "DayPicker-Footer",
            "interactionDisabled": "DayPicker--interactionDisabled",
            "month": "DayPicker-Month",
            "months": "DayPicker-Months",
            "navBar": "DayPicker-NavBar",
            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
            "outside": "outside",
            "selected": "selected",
            "today": "today",
            "todayButton": "DayPicker-TodayButton",
            "week": "DayPicker-Week",
            "weekNumber": "DayPicker-WeekNumber",
            "weekday": "DayPicker-Weekday",
            "weekdays": "DayPicker-Weekdays",
            "weekdaysRow": "DayPicker-WeekdaysRow",
            "wrapper": "DayPicker-wrapper",
          }
        }
        enableOutsideDaysClick={true}
        fixedWeeks={false}
        initialMonth={2017-12-09T22:00:00.000Z}
        labels={
          {
            "nextMonth": "Next Month",
            "previousMonth": "Previous Month",
          }
        }
        locale="en"
        localeUtils={
          {
            "formatDate": [Function],
            "formatDay": [Function],
            "formatMonthTitle": [Function],
            "formatWeekdayLong": [Function],
            "formatWeekdayShort": [Function],
            "getFirstDayOfWeek": [Function],
            "getMonths": [Function],
            "parseDate": [Function],
          }
        }
        navbarElement={
          <Navbar
            classNames={
              {
                "body": "DayPicker-Body",
                "caption": "DayPicker-Caption",
                "container": "DayPicker",
                "day": "DayPicker-Day",
                "disabled": "disabled",
                "footer": "DayPicker-Footer",
                "interactionDisabled": "DayPicker--interactionDisabled",
                "month": "DayPicker-Month",
                "months": "DayPicker-Months",
                "navBar": "DayPicker-NavBar",
                "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
                "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
                "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
                "outside": "outside",
                "selected": "selected",
                "today": "today",
                "todayButton": "DayPicker-TodayButton",
                "week": "DayPicker-Week",
                "weekNumber": "DayPicker-WeekNumber",
                "weekday": "DayPicker-Weekday",
                "weekdays": "DayPicker-Weekdays",
                "weekdaysRow": "DayPicker-WeekdaysRow",
                "wrapper": "DayPicker-wrapper",
              }
            }
            dir="ltr"
            labels={
              {
                "nextMonth": "Next Month",
                "previousMonth": "Previous Month",
              }
            }
            showNextButton={true}
            showPreviousButton={true}
          />
        }
        numberOfMonths={1}
        onDayClick={[Function]}
        pagedNavigation={false}
        renderDay={[Function]}
        renderWeek={[Function]}
        reverseMonths={false}
        selectedDays={2017-12-09T22:00:00.000Z}
        showOutsideDays={true}
        showWeekDays={true}
        showWeekNumbers={true}
        tabIndex={0}
        weekdayElement={<Weekday />}
      />
    </DatePickerWrapper>
  </Overlay>
</div>
`;
